fabric-biome-api-v1/src/main/java/net/fabricmc/fabric/impl/biome/TheEndBiomeData.java

```diff
@@ -103,26 +114,27 @@ public static class Overrides {
 
 		// cache for our own sampler (used for random biome replacement selection)
 		private final Map<MultiNoiseUtil.MultiNoiseSampler, PerlinNoiseSampler> samplers = new WeakHashMap<>();
-		// current seed, set from ChunkGenerator hook since it is not normally available
-		private static ThreadLocal<Long> seed = new ThreadLocal<>();
 
 		public Overrides(Registry<Biome> biomeRegistry) {
```

Comment:
What about saving this registry in a field and just using the static biome maps to pick a replacement?

Reply:
Could work. I didn’t refactor most of the code but I suppose it makes the stuff overall better and more performance efficient.

Reply:
Now that I think about it, storing registry entry might be more efficient because `pick` I believe is called often.